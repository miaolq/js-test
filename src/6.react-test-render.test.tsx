// const React = require('react')
// const TestRenderer = require('react-test-renderer')
import React from 'react'
import TestRenderer, { act, create } from 'react-test-renderer'

function MyComponent() {
  return (
    <div>
      <SubComponent foo="bar" />
      <p className="my">Hello</p>
    </div>
  )
}

function SubComponent(props) {
  return <p className="sub">{props.foo}</p>
}

const testRenderer = TestRenderer.create(<MyComponent />)
const json = testRenderer.toJSON()
const tree = testRenderer.toTree()
const testInstance = testRenderer.root

// testInstance.find(instance => {
//   console.dir(instance)
// })

test('SubComponent props foo should be bar', () => {
  expect(testInstance.findByType(SubComponent).props.foo).toBe('bar')

  expect(testInstance.findByProps({ className: 'sub' }).children).toEqual([
    'bar',
  ])
})

test('create act', () => {
  let root
  act(() => {
    root = create(<SubComponent foo={1} />)
  })

  expect(root.toJSON()).toMatchSnapshot()

  act(() => {
    root.update(<SubComponent foo={2} />)
  })

  expect(root.toJSON()).toMatchSnapshot()
})
