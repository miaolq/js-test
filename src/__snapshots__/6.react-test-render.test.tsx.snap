// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create act 1`] = `
<p
  className="sub"
>
  1
</p>
`;

exports[`create act 2`] = `
<p
  className="sub"
>
  2
</p>
`;
